src/modules/UI/scenes/SendConfirmation/action.js

```diff
@@ -1,82 +1,88 @@
 
-export const UPDATE_AMOUNT_SATOSHI = PREFIX + 'UPDATE_AMOUNT_SATOSHI'
 export const UPDATE_LABEL = PREFIX + 'UPDATE_LABEL'
-export const UPDATE_MAX_AVAILABLE_TO_SPEND_IN_CRYPTO = PREFIX + 'UPDATE_MAX_AVAILABLE_TO_SPEND_IN_CRYPTO'
-export const ENABLE_SLIDER = PREFIX + 'ENABLE_SLIDER'
-export const UPDATE_DRAFT_STATUS = PREFIX + 'UPDATE_DRAFT_STATUS'
 export const UPDATE_IS_KEYBOARD_VISIBLE = PREFIX + 'UPDATE_IS_KEYBOARD_VISIBLE'
-export const UPDATE_FEE = PREFIX + 'UPDATE_FEE'
-export const UPDATE_MAX_SATOSHI = PREFIX + 'UPDATE_MAX_SATOSHI'
 export const UPDATE_SPEND_PENDING = PREFIX + 'UPDATE_SPEND_PENDING'
-
-export const UPDATE_WALLET_TRANSFER = PREFIX + 'UPDATE_WALLET_TRANSFER'
-export const UPDATE_PUBLIC_ADDRESS = PREFIX + 'UPDATE_PUBLIC_ADDRESS'
-export const UPDATE_SPEND_INFO = PREFIX + 'UPDATE_SPEND_INFO'
 export const RESET = PREFIX + 'RESET'
-
-export const UPDATE_CRYPTO_AMOUNT_REQUEST = PREFIX + 'UPDATE_CRYPTO_AMOUNT_REQUEST'
-export const USE_MAX_CRYPTO_AMOUNT = PREFIX + 'USE_MAX_CRYPTO_AMOUNT'
 export const UPDATE_PARSED_URI = PREFIX + 'UPDATE_PARSED_URI'
 export const UPDATE_TRANSACTION = PREFIX + 'UPDATE_TRANSACTION'
 
-export const UPDATE_NATIVE_AMOUNT = PREFIX + 'UPDATE_NATIVE_AMOUNT'
-export const CHANGE_MINING_FEE = PREFIX + 'CHANGE_MINING_FEE'
+export const updateWalletTransfer = (wallet: AbcCurrencyWallet) =>
+  (dispatch: any) => dispatch(updateLabel(wallet.name))
+
+export const updateParsedURI = (parsedUri: AbcParsedUri) =>
+  async (disptach: any) => disptach(createTX({ ...parsedUri }))
 
-export type UpdateTransactionAction = {
-  type: typeof UPDATE_TRANSACTION,
-  data: {
-    transaction?: AbcTransaction | null,
-    error?: Error | null
-  }
+export const updateMiningFees = (networkFeeOption: string, customNetworkFee: any) =>
+  async (disptach: any) => disptach(createTX({ networkFeeOption, customNetworkFee }))
+
+export const updateMaxSpend = () => async (dispatch: any, getState: any) => {
+  const state = getState()
+  const walletId = getSelectedWalletId(state)
+  const abcWallet = getWallet(state, walletId)
+  const spendInfo = getSpendInfo(state)
+  getMaxSpendable(abcWallet, spendInfo)
+  .then(nativeAmount => dispatch(updateParsedURI({ nativeAmount })))
+  .catch(e => console.log(e))
 }
 
-export const updateAmountSatoshi = (amountSatoshi: string) => ({
-  type: UPDATE_AMOUNT_SATOSHI,
-  data: {amountSatoshi}
+export const createTX = (newParsedUri: AbcMakeSpendInfo) => async (dispatch: any, getState: any) => {
+  const state = getState()
+  const walletId = getSelectedWalletId(state)
+  const abcWallet = getWallet(state, walletId)
+  const spendInfo = getSpendInfo(state, newParsedUri)
+  makeSpend(abcWallet, spendInfo)
+  .then(abcTransaction => dispatch(updateTransaction(abcTransaction, null)))
+  .then(() => dispatch(makeParsedURI({ ...newParsedUri })))
+  .catch(e => dispatch(updateTransaction(null, e)))
+}
+
+export const makeParsedURI = (parsedUri: AbcParsedUri) => ({
```

Comment:
This is miss-typed. You actually are using an `AbcMakeSpendInfo` not an `AbcParsedUri` which should really be called a `GuiMakeSpendInfo` since this is not an ABC type but a GUI specific type. Change all references to `makeSpendInfo` including `UPDATE_PARSED_URI` to `MAKE_SPEND_INFO`

Reply:
fixed in https://github.com/Airbitz/edge-react-gui/pull/389